### PR TITLE
Display butchery duration depending on available tools

### PR DESCRIPTION
**Pull request by mqrause**
#### Summary

SUMMARY: Bugfixes "Butchery menu displays required time depending on available tools"

#### Purpose of change

This needed an update to reflect the changes in recent butchery PRs.

#### Describe the solution

Multiply the displayed time with the appropriate factor from butchery requirements.

#### Testing

Tested with the changes from #42837. Open butcher menu while fulfilling different requirements.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -8437,10 +8438,18 @@ static void butcher_submenu( const std::vector<map_stack::iterator> &corpses, in
     auto cut_time = [&]( butcher_type bt ) {
         int time_to_cut = 0;
         if( corpse != -1 ) {
-            time_to_cut = butcher_time_to_cut( player_character, *corpses[corpse], bt );
+            const mtype &item = *( *corpses[corpse] ).get_mtype();
```
Comment (KorGgenT) [inline src/game.cpp:8441]: calling a variable `item` hides the `item` class. if the LGTM check doesn't catch it some time i'd be surprised.

Comment (mqrause) [reply src/game.cpp:8441]: Oh god, yeah, I shouldn't code at night.

**Comment by cjjp** [comment]
@mqrause 
Been getting segfault heaps when I butcher since this change. It started after I installed scalpels and happens regardless of power or if the scalpels are enabled. It only stops happening if I'm carrying a scalpel in inventory.
Yet to reproduce it on a fresh character so I'm holding up creating an issue for it.
I'm just assuming it's related to this .

 https://pastebin.com/depSMEfG

**Comment by mqrause** [comment]
Are you sure it's not fixed by #43152?

**Comment by cjjp** [comment]
It probably is. Not sure how I missed the issues.

**Pull request merged**